infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/aws/StackCreationValidator.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.testgrid.infrastructure.providers.aws;
+
+import com.amazonaws.services.cloudformation.AmazonCloudFormation;
+import com.amazonaws.services.cloudformation.model.DescribeStackEventsRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStackEventsResult;
+import com.amazonaws.services.cloudformation.model.DescribeStacksRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStacksResult;
+import com.amazonaws.services.cloudformation.model.StackEvent;
+import com.amazonaws.services.cloudformation.model.StackStatus;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.testgrid.common.util.StringUtil;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Validates the Stack creation in AWS.
+ */
+public class StackCreationValidator {
+
+    private static final Logger logger = LoggerFactory.getLogger(StackCreationValidator.class);
+
+    /**
+     * Periodically checks for the status of stack creation until a defined timeout.
+     *
+     * @param timeout         Time to wait upon the desired response.
+     * @param timeoutTimeUnit TimeUnit of timeout value.
+     * @param pollInterval    Time interval to perform polling.
+     * @param pollTimeUnit    TimeUnit of pollInterval.
+     */
+    public void waitForStack(int timeout, TimeUnit timeoutTimeUnit, int pollInterval,
+                                  TimeUnit pollTimeUnit, String stackName, AmazonCloudFormation cloudFormation)
+            throws ConditionTimeoutException {
+        Awaitility.with().pollInterval(pollInterval, pollTimeUnit).await().
+                atMost(timeout, timeoutTimeUnit).until(isStackCreationSuccessful(stackName, cloudFormation));
```

Comment:
What's the behavior here if AWSCallable#call threw a TestGridInfrastructureException? 

As I see, TestGridInfrastructureException is thrown if the AWS stack creation is not successful. In that case, we have to stop waiting for the stack, yes?

Reply:
Added TestGridInfrastructureException exception to the method signature which will be ultimately caught by RunTestPlan#provisionInfrastructure.